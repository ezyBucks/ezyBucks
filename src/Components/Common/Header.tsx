import * as React from 'react';
import { Button } from 'antd';
import { Link } from 'react-router-dom';

/**
 * Class to render the header for the application
 */
export const Header: React.SFC = () => {
    return (
        <header className="App-header">
            <span style={{ marginRight: 'auto' }}>ezyBucks</span>
            <Button type="primary" style={{ margin: '2px' }}>
                <Link to="/">Home</Link>
            </Button>
            <Button type="primary" style={{ margin: '2px' }}>
                <Link to="/login">Login</Link>
            </Button>
            <Button type="primary" style={{ margin: '2px' }}>
                <Link to="/register">Register</Link>
            </Button>
            <Button type="primary" style={{ margin: '2px' }}>
                <Link to="/private">private</Link>
            </Button>
            <Button type="primary" style={{ margin: '2px' }}>
                <Link to="/public">public</Link>
            </Button>
        </header>
    );
};
