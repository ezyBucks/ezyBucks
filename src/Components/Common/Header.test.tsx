import React from 'react';
import renderer from 'react-test-renderer';
import { Header } from './Header';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router';

it('renders without crashing', () => {
    shallow(<Header />);
});

it('fully renders without crashing', () => {
    const wrapper = mount(
        <MemoryRouter>
            <Header />
        </MemoryRouter>
    );
    expect(wrapper.contains(<Header />)).toBe(true);
});

it('matches the snapshot', () => {
    const header = renderer
        .create(
            <MemoryRouter>
                <Header />
            </MemoryRouter>
        )
        .toJSON();
    expect(header).toMatchSnapshot();
});
