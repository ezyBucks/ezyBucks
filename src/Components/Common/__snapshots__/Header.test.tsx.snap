// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<header
  className="App-header"
>
  <span
    style={
      Object {
        "marginRight": "auto",
      }
    }
  >
    ezyBucks
  </span>
  <button
    className="ant-btn ant-btn-primary"
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
      }
    }
    type="button"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
  </button>
  <button
    className="ant-btn ant-btn-primary"
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
      }
    }
    type="button"
  >
    <a
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
  </button>
  <button
    className="ant-btn ant-btn-primary"
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
      }
    }
    type="button"
  >
    <a
      href="/register"
      onClick={[Function]}
    >
      Register
    </a>
  </button>
  <button
    className="ant-btn ant-btn-primary"
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
      }
    }
    type="button"
  >
    <a
      href="/private"
      onClick={[Function]}
    >
      private
    </a>
  </button>
  <button
    className="ant-btn ant-btn-primary"
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
      }
    }
    type="button"
  >
    <a
      href="/public"
      onClick={[Function]}
    >
      public
    </a>
  </button>
</header>
`;
